src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -77,34 +72,29 @@ const MyDiagnoses = () => {
   return (
     <Background>
       <View style={styles.container}>
-        <ForceRerenderOnNavigation resetStateFunction={refetchDiagnoses} />
         <Header isScrolling={scrolling} />
-        {downloadingDiagnoses && <ActivityIndicator size='large' />}
-        {diagnoses &&
-          <FlatList
-            ref={flatListRef}
-            data={diagnoses}
-            showsVerticalScrollIndicator={false}
-            renderItem={({ item }) => item}
-            keyExtractor={item => String(diagnoses.indexOf(item))}
-            onRefresh={getRenderedDiagnoses}
-            refreshing={refreshing}
-            disableVirtualization={false}
-            onScrollBeginDrag={() => setScrolling(true)}
-            onScrollEndDrag={({ nativeEvent }) => {
-              const snapToTopIfScrollingNearTheFirstElement = () => {
-                if (nativeEvent.contentOffset.y < OFFSET_THRESHOLD_TO_SNAP_FLATLIST) {
-                  flatListRef.current.scrollToIndex({ animated: true, index: 0 })
-                  setScrolling(false)
-                } else {
-                  setScrolling(true)
-                }
+        <FlatList
+          ref={flatListRef}
+          data={diagnoses.current}
+          showsVerticalScrollIndicator={false}
+          renderItem={({ item }) => item.renderedDiagnose}
```

Comment:
I don't like much this pattern of pre-rendering. I don't know if it plays nicely with React's internals (reconciliation in particular, automatic memory management [we're keeping all the diagnoses rendered in-memory and preventing garbage-collection]). Instead of a function `renderDiagnose` and storing the diagnose here, you could just have a component `DiagnoseListItem` that receives an `item` prop, and pass that component here. Example:
```javascript
// in this function:
  <FlatList
    // ...
   renderItem={Diagnose}
   // ...
  />
  // ...

function Diagnose ({ item }) {
  return // render the diagnose here
}
```

What do you guys think? cc @MPucheta

Reply:
Mmm. I think we can save some time of rendering if we actually save pre-rendered objects but obviously it uses more memory. Right now there is no big impact on memory because there are few diagnoses.  In a future having +50 diagnoses could change this. I think we should measure it to see what is better...

Reply:
Seems that pre-rendering like this doesn't save any time on rendering so I'm making the change.